### PR TITLE
LPS-153134 Test Autom. StructuredContentAssetPriority#StructuredContentDraftContainsModifiedPriorityValue

### PR DESCRIPTION
**Pull request by mateusralv**
**Background**
Create automation test StructuredContentAssetPriority#StructuredContentDraftContainsModifiedPriorityValue

**Test Plan**
**Given** a content structure with a content-field of dataType "string" and label "content" and name "content" created in the portal
**And Given** with Headless Admin Content POST "/v1.0/sites/{siteId}/structured-contents/draft" a structured content draft "content1" with a priority 1.0 set is created
**When** with Headless Delivery PUT "/v1.0/structured-contents/{structuredContentId}" a structured content draft "content1" priority is modified to 3.0
**And When** with Headless Admin Content GET "/v1.0/sites/{siteId}/structured-contents" and filter set to "priority eq 3.0"
**Then** I should see in response "content1" with the updated priority value

**JIRA Ticket:**
https://issues.liferay.com/browse/LPS-153134

### CODE REVIEW

**Comment by mateusralv** [comment]
ci:test:frontend

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessAdminContentAPI/SupportContentVersioning/StructuredContentAssetPriority.testcase [hunk 1]
```diff
@@ -33,6 +33,44 @@ definition {
...
+		}
+
+		task ("When web contents are filtered with 3.0 priority") {
+			var response = HeadlessWebcontentAPI.filterStructuredContent(filtervalue = "priority%20eq%203.0");
+		}
+
+		task ("Then I should see in response content1 with the updated priority value") {
+			HeadlessWebcontentAPI.assertStructuredContentIdIsFilteredWithCorrectValue(
+				editStructuredContentId = "${editStructuredContentId}",
+				responseToParse = "${response}");
```
Comment (magjed4289) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessAdminContentAPI/SupportContentVersioning/StructuredContentAssetPriority.testcase:70]: we want to check the priority value here, maybe we could show it by extracting it from the body response and assert whether the result is the expected one?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessAdminContentAPI/SupportContentVersioning/StructuredContentAssetPriority.testcase [hunk 2]
```diff
@@ -33,6 +33,44 @@ definition {
 		}
 	}
 
+	@priority = "5"
+	test StructuredContentDraftContainsModifiedPriorityValue {
+		property portal.acceptance = "true";
+
+		task ("When in Headless Admin Content i create a structuredContent draft with with a priority field set") {
```
[the remark this answers is not in the document]
Comment (magjed4289) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessAdminContentAPI/SupportContentVersioning/StructuredContentAssetPriority.testcase:40]: also, can we make "i" to be "I"? :D

**Review by magjed4289 — CHANGES_REQUESTED** [review]

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessAdminContentAPI/SupportContentVersioning/StructuredContentAssetPriority.testcase (hunk 1 above)
Comment (mateusralv) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessAdminContentAPI/SupportContentVersioning/StructuredContentAssetPriority.testcase]: Done!

**Comment by mateusralv** [comment]
Hey @magjed4289, I made the requested changes, does it look good? 🤔

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessAdminContentAPI/SupportContentVersioning/StructuredContentAssetPriority.testcase [hunk 3]
```diff
@@ -33,6 +33,44 @@ definition {
...
+			var responseToParse = HeadlessWebcontentAPI.createStructuredContentDraft(
+				data = "Text",
+				ddmStructureId = "${ddmStructureId}",
+				label = "Text",
+				name = "Text",
+				priority = "1",
+				title = "Title");
+		}
+
+		task ("When edit a web content with different priorities") {
```
Comment (magjed4289) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessAdminContentAPI/SupportContentVersioning/StructuredContentAssetPriority.testcase:52]: Maybe we can change it to `When a structured content draft priority is modified` or something similar with priorities in singular? :)

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessAdminContentAPI/SupportContentVersioning/StructuredContentAssetPriority.testcase [hunk 4]
```diff
@@ -59,7 +97,7 @@ definition {
 	test StructuredContentDraftContainsSetPriorityValue {
 		property portal.acceptance = "true";
 
-		task ("When in Headless Admin Content i create a structuredContent draft with with a priority field set") {
+		task ("When in Headless Admin Content I create a structuredContent draft with with a priority field set") {
```
Comment (magjed4289) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessAdminContentAPI/SupportContentVersioning/StructuredContentAssetPriority.testcase:100]: repeated 'with', can you please remove it? Thank you!

**Review by magjed4289 — APPROVED** [review]
Small changes indicated, once implemented the code is ready to being forward :) Thank you!

**Comment by mateusralv** [comment]
> Small changes indicated, once implemented the code is ready to being forward :) Thank you!

Changes made!

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessAdminContentAPI/SupportContentVersioning/StructuredContentAssetPriority.testcase [hunk 5]
```diff
@@ -33,6 +33,44 @@ definition {
 		}
 	}
 
+	@priority = "5"
+	test StructuredContentDraftContainsModifiedPriorityValue {
+		property portal.acceptance = "true";
+
+		task ("Given in Headless Admin Content I create a structuredContent draft with with a priority field set") {
```
Comment (magjed4289) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessAdminContentAPI/SupportContentVersioning/StructuredContentAssetPriority.testcase:40]: there's double "with" in this line too 😅

**Review by magjed4289 — APPROVED** [review]
after removing the typo, ready to forward! :)

**Review by magjed4289 — APPROVED** [review]

**Comment by magjed4289** [comment]
ci:forward

**Comment by mateusralv** [comment]
Hi @magjed4289 , I noticed that SF was crashing and I saw that it was happening because a macro had its name changed. Updates have already been made!

**Comment by mateusralv** [comment]
ci:test:sf

**Comment by mateusralv** [comment]
ci:test:headless-frontend-functional:nocompile

**Comment by mateusralv** [comment]
ci:forward

**Pull request closed without merging**